bsp/Src/BSP_SPI.c

```diff
@@ -0,0 +1,196 @@
+#include "BSP_SPI.h"
+
+#define LOCAL_BUFFER_SIZE 128 // Modify if needed
+static uint8_t localTxMetaBuffer[LOCAL_BUFFER_SIZE];
+static StaticQueue_t localTxMetaStatic;
+static QueueHandle_t localTxMetaQueue;
+static uint8_t localRxMetaBuffer[LOCAL_BUFFER_SIZE];
+static StaticQueue_t localRxMetaStatic;
+static QueueHandle_t localRxMetaQueue;
+
+static QueueHandle_t* _transmitQueuePtr;
+static QueueHandle_t* _receiveQueuePtr;
+
+/**
+ * @brief Initialize the provided SPI port
+ * 
+ * @param spiHandle The SPI handle to configure and init
+ * @return HAL_StatusTypeDef 
+ */
+HAL_StatusTypeDef BSP_SPI_Init(SPI_HandleTypeDef* spiHandle, QueueHandle_t* txPtr, QueueHandle_t* rxPtr) {
+    HAL_StatusTypeDef stat;
+
+    // assign local pointers
+    // and create metadata queues
+    _transmitQueuePtr = txPtr;
+    _receiveQueuePtr = rxPtr;
+    localTxMetaQueue = xQueueCreateStatic(LOCAL_BUFFER_SIZE, sizeof(uint8_t), localTxMetaBuffer, &localTxMetaStatic);
+    localRxMetaQueue = xQueueCreateStatic(LOCAL_BUFFER_SIZE, sizeof(uint8_t), localRxMetaBuffer, &localRxMetaStatic);
+    
+    // We are only allowing SPI2/3
+    if (spiHandle->Instance == SPI2) {
+        if (__HAL_RCC_SPI2_IS_CLK_DISABLED()) {
+            __HAL_RCC_SPI2_CLK_ENABLE();
+        }
+    }
+    else if (spiHandle->Instance == SPI3) {
+        if (__HAL_RCC_SPI3_IS_CLK_DISABLED()) {
+            __HAL_RCC_SPI3_CLK_ENABLE();
+        }
```

Comment:
why no SPI1? Make sure the STM32L431CBT6 doesn't have more than 3 SPIs (it shouldn't).